app/src/main/kotlin/com/wire/android/notification/WireFirebaseMessagingService.kt

```diff
@@ -0,0 +1,39 @@
+package com.wire.android.notification
+
+import com.google.firebase.messaging.FirebaseMessagingService
+import com.google.firebase.messaging.RemoteMessage
+import com.wire.android.di.KaliumCoreLogic
+import com.wire.kalium.logic.CoreLogic
+import dagger.hilt.android.AndroidEntryPoint
+import kotlinx.coroutines.runBlocking
+import javax.inject.Inject
+
+@AndroidEntryPoint
+class WireFirebaseMessagingService : FirebaseMessagingService() {
+
+    @Inject
+    @KaliumCoreLogic
+    lateinit var coreLogic: CoreLogic
+
+    @Inject
+    lateinit var wireNotificationManager: WireNotificationManager
+
+    override fun onNewToken(p0: String) {
+        super.onNewToken(p0)
+        coreLogic.getAuthenticationScope().saveNotificationToken(p0, "GCM")
+    }
+
+    override fun onMessageReceived(message: RemoteMessage) {
+        super.onMessageReceived(message)
+        var userId = ""
+        for (items in message.data) {
+            if (items.key == "user") {
+                userId = items.value
+                break
+            }
+        }
+        runBlocking {
```

Comment:
Did you try something like [that ](https://stackoverflow.com/questions/63405673/how-to-call-suspend-function-from-service-android) instead of `runBlocking` ?

Reply:
yes this is the one that I mentioned I tried and it was stopping the job from getting new notification , but I will try it again , because that one might be because of the delay that we mentioned in the morning

Reply:
yeah it's still the same , it's not launching